applications/nrf_desktop/src/modules/ble_state.c

```diff
@@ -19,15 +19,33 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_BLE_STATE_LOG_LEVEL);
 
+#ifdef CONFIG_BT_PERIPHERAL
+#define SECURITY_FAIL_TIMEOUT_MS \
+	K_SECONDS(CONFIG_DESKTOP_BLE_SECURITY_FAIL_TIMEOUT_S)
+#else
+#define SECURITY_FAIL_TIMEOUT_MS 0
+#endif
 
 struct bond_find_data {
 	bt_addr_le_t peer_address;
 	u8_t peer_id;
 	u8_t peer_count;
 };
 
+static struct k_delayed_work security_timeout;
+static struct bt_conn *active_conn[CONFIG_BT_MAX_CONN];
 
-static const struct bt_conn *active_conn[CONFIG_BT_MAX_CONN];
+
+static void security_timeout_fn(struct k_work *w)
+{
+	static_assert(IS_ENABLED(CONFIG_BT_CENTRAL) ||
```

Comment:
although we don't use it one device can be both central and peripheral
It would be better to explicitly check for peripheral i.e.
`assert(!peripheral || x)`

Reply:
done